src/solvers/dgsem_unstructured/dg_2d.jl

```diff
@@ -368,7 +369,8 @@ end
 # inlined version of the boundary flux calculation along a physical interface where the
 # boundary flux values are set according to a particular `boundary_condition` function
 @inline function calc_boundary_flux!(surface_flux_values, t, boundary_condition,
-                                     mesh::UnstructuredMesh2D, equations,
+                                     mesh::UnstructuredMesh2D,
+                                     nonconservative_terms::Val{false}, equations,
```

Comment:
For clarification, is this here because the non-conservative terms don't factor into the imposition of boundary conditions?

Edit: I think @ranocha has a similar question below. Feel free to resolve this m, I'll follow that comment.

Reply:
This version, when `nonconservative_terms` is false, would be for equation systems in purely conservative form. For the shallow water equations the `jump_b` part does not contribute to the boundary flux because the bottom is assumed to be continuous. However, we still must include the diagonal (consistent) part of the bottom topography gradient at the boundary, because it have been "removed" from the split form volume kernel as the matrix `derivative_split` is zero all along the diagonal. These terms have moved on the surface.

This is very particular to the shallow water equations